demisto_sdk/commands/unify_packs/packs_unifier.py

```diff
@@ -0,0 +1,191 @@
+
+
+class PacksManager(ArtifactsManager):
+    def __init__(self, pack_paths: str, all_in_one_zip: bool, quite_mode: bool, **kwargs):
+        super().__init__(**kwargs)
+        self.init_packs(pack_paths)
+        self.zip_all = all_in_one_zip
+        self.quite_mode = quite_mode
+        self.output_path = f'{self.content_uploadable_zips_path}.zip' if self.zip_all \
+            else str(self.content_uploadable_zips_path)
+
+    def init_packs(self, pack_paths):
+        """
+
+        Args:
+            pack_paths: CSV str with the pack paths
+
+        init dict that map pack name to Pack object and init also the pack_names property
+
+        """
+        self.packs = {}
+        for path_str in arg_to_list(pack_paths):
+            path = Path(path_str)
+            if len(path.parts) == 2 and path.parts[0] == PACKS_DIR:  # relative path from Packs/...
+                path = self.content.path / path
+
+            if not os.path.exists(path):
+                click.secho(f'Error: Given input path: {path} does not exist, ignored', fg='bright_red')
+                continue
+
+            self.packs.update({path.name: Pack(path)})
+
+        self.pack_names = [*self.packs]
+
+    def get_relative_pack_path(self, content_object: ContentObject):
+        """
+
+        Args:
+            content_object: the object to get the relative path for
+
+        Returns:
+            the path of the given object relative from the pack directory, for example HelloWorld/Scripts
+
+
+        """
+        for part in content_object.path.parts:
+            if part in self.packs:
+                return content_object.path.relative_to(self.packs[part].path.parent)
+
+    def get_base_path(self):
+        """
+
+        Returns:
+            None, as packs can came outside from Content directory
+
+        """
+        return None
+
+    def get_dir_to_delete(self):
+        """
+
+        Returns:
+            the value of ArtifactsManager.get_dir_to_delete and if zip_all is True - also the uploadable_packs dir
+        """
+
+        result = super().get_dir_to_delete()
+        if self.zip_all:
+            result.append(self.content_uploadable_zips_path)
+        return result
+
+    def dump_packs(self):
+        """
+
+        Dump all the packs stored in the packs field
+        and will print summery according to the quit_mode field
+
+        """
+        reports = []
+        # we quite the outputs and in case we want the output - a summery will be printed
+        with QuiteModeController(quite_logger=True, quite_output=True), PacksDirsHandler(self):
+            for pack_name in self.pack_names:
+                if pack_name not in IGNORED_PACKS:
+                    reports.append(dump_pack(self, self.packs[pack_name]))
+
+        if not self.quite_mode:
+            logger = logging.getLogger('demisto-sdk')
+            for report in reports:
+                logger.info(report.to_str(self.get_base_path()))
+
+
+class QuiteModeController:
```

Comment:
@roysagi 
Please give your opinion on it.
If you think that's a good solution then maybe we should move it to `tools.py`?